### PR TITLE
Add views to presenting row-based layouts

### PR DESCRIPTION
**Pull request by Supereg**
# Add views to presenting row-based layouts

## :recycle: Current situation & Problem
We identified a use case providing a default component to present `ListRow`s that automatically adapt the the current dynamic type size, device orientation and size classes.
This PR adds a new `DynamicHStack` that automatically adapts row-based, horizontal content to the current dynamic type size.

## :gear: Release Notes 
* Added `DynamicHStack` for presenting row-based content that adapts to the current dynamic type size.
* Added the `ListRow` view to present key-value-based list rows, based on the `DynamicHStack`.

## :books: Documentation
Documentation and code examples were provided. I restructured the landing page a bit to accommodate for those new elements.

## :white_check_mark: Testing
This PR adds basic support for snapshot testing using baseline images. These allow to unit test view layouts across builds. We rely on the [SnapshotTesting](https://github.com/pointfreeco/swift-snapshot-testing) framework for that.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .reuse/dep5
Comment (PSchmiedmayer) [inline .reuse/dep5:1]: Nice, thank you for the addition here!

File: Tests/SpeziViewsTests/SnapshotTests.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SnapshotTesting
```
Comment (PSchmiedmayer) [inline Tests/SpeziViewsTests/SnapshotTests.swift:9]: Nice; cool addition to the repo and what we can test!

File: Sources/SpeziViews/Resources/Localizable.xcstrings
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Resources/Localizable.xcstrings:1]: Thank you!

**Review by PSchmiedmayer — APPROVED** [review]
@Supereg Thank you for the changes! The PR looks good, only bigger question would be the existence of the `NoInformationText` view, let me know what you think!

File: Sources/SpeziViews/Views/Text/NoInformationText.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+///             // ...
+///         }
+///     }
+///
+///     init(_ contacts: [Contact]) {
+///         self.contacts = contacts
+///     }
+/// }
+/// ```
+public struct NoInformationText<Header: View, Caption: View>: View {
```
[the remark this answers is not in the document]
Comment (Supereg) [reply Sources/SpeziViews/Views/Text/NoInformationText.swift]: You are totally right. This thing already exists in SwiftUI. I didn't know that back then when I needed that in NAMS. I migrated to the `ContentUnavailableView` and it looks really good 🚀 Thanks for the hint. Removed the `NoInformationText` view.

File: Tests/SpeziViewsTests/SnapshotTests.swift (hunk 1 above)
Comment (Supereg) [reply Tests/SpeziViewsTests/SnapshotTests.swift:9]: Yes, pretty cool to test layout. Also very easy to use and create baseline snapshots. Seems to have some widespread use out there 🚀 
Might want to check what other repositories we can use this in.

File: .reuse/dep5
Comment (Supereg) [reply .reuse/dep5:1]: Yes, though this was easier for this dynamic snapshots directory 👍

File: Sources/SpeziViews/Views/Text/NoInformationText.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (PSchmiedmayer) [reply Sources/SpeziViews/Views/Text/NoInformationText.swift]: Thank you! 🎉

**Pull request merged**